### PR TITLE
protect balance parameters by setting legal range(#23454)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #23454

### CODE REVIEW

File: internal/querycoordv2/checkers/controller.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func (controller *CheckerController) Start(ctx context.Context) {
 				controller.check(ctx)
 
 			case <-controller.checkCh:
+			case <-controller.scheduler.IsEmpty():
```
Comment (yah01) [inline internal/querycoordv2/checkers/controller.go:100]: need `fallthrough` or the `checkCh` will not work

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/querycoordv2/checkers/controller.go (hunk 1 above)
Comment (MrPresent-Han) [reply internal/querycoordv2/checkers/controller.go:100]: right, I will reconstruct this logic

File: internal/util/paramtable/component_param.go [hunk 2]
```diff
@@ -987,6 +987,14 @@ func (p *queryCoordConfig) initGlobalRowCountFactor() {
 	if err != nil {
 		panic(err)
 	}
+	if globalRowCountFactor > 1.0 {
+		log.Warn("globalRowCountFactor should not be more than 1.0, force set to 1.0")
+		globalRowCountFactor = 1.0
+	}
+	if globalRowCountFactor <= 0 {
```
Comment (congqixia) [inline internal/util/paramtable/component_param.go:994]: shall this be `globalRowCountFactor <= 0.1` ?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**